btree.go

```diff
@@ -64,6 +64,42 @@ type Item interface {
 	Less(than Item) bool
 }
 
+// FreeList represents a free list of btree nodes. By default each
+// BTree has its own FreeList, but multiple BTrees can share the same
+// FreeList.
+type FreeList struct {
+	freelist []*node
+}
+
+func NewFreeList() *FreeList {
```

Comment:
Pass in size, and use DefaultFreeListSize in New(...)

    var DefaultFreeListSize = 32

    func NewFreeList(size int) *FreeList {...}

    func New(degree int) { return NewWithFreeList(degree, NewFreeList(DefaultFreeListSize)) }